encrypted-media-respec.html

```diff
@@ -3025,6 +3016,53 @@
       <h3>Implementation Requirements</h3>
       <p>This section defines implementation requirements - for both user agents and <a def-id="keysystems"></a>, including the <a def-id="cdm"></a> and server - that may not be explicitly addressed in the algorithms.</p>
 
+	    <section id="cdm-constraint-requirements">
+        <h3>CDM Constraints</h3>
+        <p>
+          User agent implementers MUST ensure that CDMs do not access any information, storage or system capabilities that are not reasonably required for playback of protected media using the features of this specification. 
+          Specifically, the CDM SHALL NOT:
+        </p>
+        <ul>
+          <li>
+            <p>
+              Access network resources, either local or remote, except via the user agent, or as part of the user agent, as explicitly permitted by this specification.
+            </p>
+          </li>
+          <li>
+            <p>
+              Access storage (disk or memory), except where reasonably required for playback of protected media using the features of this specification.
+            </p>
+          </li>
+          <li>
+            <p>
+              Access user data other than CDM state and persistent data.
+            </p>
+          </li>
+          <li>
+            <p>
+              Access hardware components or devices, except where reasonably required for playback of protected media using the features of this specification.
+            </p>
+          </li>
+        </ul>
+        <p>
+          User Agent implementers may use various techniques to meet the above requirements. For example, a User Agent implementer also implementing their own CDM may include the above as design requirements for that component. 
+          A User Agent implementer making use of a third party CDM may ensure that it executes in a constrained environment (e.g., "sandbox") without access to the prohibited information and components.
+        </p>
+      </section>  
+
+	    <section id="messaging-requirements">
+        <h3>Messaging</h3>
```

Comment:
"Messaging" sounds a bit weird and could be more limiting than intended. Perhaps we should use "Messages and Communication" as in the first sentence of the section.